From 0f9545a6afb5af11ee0b8f27f3dcb14574e6b179 Mon Sep 17 00:00:00 2001
From: Stephen Hines <srhines@google.com>
Date: Tue, 29 Sep 2015 03:57:47 -0700
Subject: [PATCH 1/2] Remove invalid clang ifdef about __builtin_isnan() and
 clang >= 3.7.

Bug: 23396112

Clang continues to have "int __builtin_isnan()" as the official builtin
definition, which causes conflicts with the NDK headers. This change
unconditionally leaves __builtin_isnan() out of the NDK headers when
using clang, since it really isn't necessary in any case.

Change-Id: Ifb5c1db40c11eb5984e1d64e8540e8a6537900ab
Signed-off-by: sub77 <sub77@ymail.com>
---
 9/platforms/android-12/arch-arm/usr/include/math.h    | 4 ++--
 9/platforms/android-12/arch-mips/usr/include/math.h   | 4 ++--
 9/platforms/android-12/arch-x86/usr/include/math.h    | 4 ++--
 9/platforms/android-13/arch-arm/usr/include/math.h    | 4 ++--
 9/platforms/android-13/arch-mips/usr/include/math.h   | 4 ++--
 9/platforms/android-13/arch-x86/usr/include/math.h    | 4 ++--
 9/platforms/android-14/arch-arm/usr/include/math.h    | 4 ++--
 9/platforms/android-14/arch-mips/usr/include/math.h   | 4 ++--
 9/platforms/android-14/arch-x86/usr/include/math.h    | 4 ++--
 9/platforms/android-15/arch-arm/usr/include/math.h    | 4 ++--
 9/platforms/android-15/arch-mips/usr/include/math.h   | 4 ++--
 9/platforms/android-15/arch-x86/usr/include/math.h    | 4 ++--
 9/platforms/android-16/arch-arm/usr/include/math.h    | 4 ++--
 9/platforms/android-16/arch-mips/usr/include/math.h   | 4 ++--
 9/platforms/android-16/arch-x86/usr/include/math.h    | 4 ++--
 9/platforms/android-17/arch-arm/usr/include/math.h    | 4 ++--
 9/platforms/android-17/arch-mips/usr/include/math.h   | 4 ++--
 9/platforms/android-17/arch-x86/usr/include/math.h    | 4 ++--
 9/platforms/android-18/arch-arm/usr/include/math.h    | 4 ++--
 9/platforms/android-18/arch-mips/usr/include/math.h   | 4 ++--
 9/platforms/android-18/arch-x86/usr/include/math.h    | 4 ++--
 9/platforms/android-19/arch-arm/usr/include/math.h    | 4 ++--
 9/platforms/android-19/arch-mips/usr/include/math.h   | 4 ++--
 9/platforms/android-19/arch-x86/usr/include/math.h    | 4 ++--
 9/platforms/android-21/arch-arm/usr/include/math.h    | 4 ++--
 9/platforms/android-21/arch-arm64/usr/include/math.h  | 4 ++--
 9/platforms/android-21/arch-mips/usr/include/math.h   | 4 ++--
 9/platforms/android-21/arch-mips64/usr/include/math.h | 4 ++--
 9/platforms/android-21/arch-x86/usr/include/math.h    | 4 ++--
 9/platforms/android-21/arch-x86_64/usr/include/math.h | 4 ++--
 9/platforms/android-3/arch-arm/usr/include/math.h     | 4 ++--
 9/platforms/android-4/arch-arm/usr/include/math.h     | 4 ++--
 9/platforms/android-5/arch-arm/usr/include/math.h     | 4 ++--
 9/platforms/android-8/arch-arm/usr/include/math.h     | 4 ++--
 9/platforms/android-9/arch-arm/usr/include/math.h     | 4 ++--
 9/platforms/android-9/arch-mips/usr/include/math.h    | 4 ++--
 9/platforms/android-9/arch-x86/usr/include/math.h     | 4 ++--
 37 files changed, 74 insertions(+), 74 deletions(-)

diff --git a/9/platforms/android-12/arch-arm/usr/include/math.h b/9/platforms/android-12/arch-arm/usr/include/math.h
index 5730cd4..16df2f1 100644
--- a/9/platforms/android-12/arch-arm/usr/include/math.h
+++ b/9/platforms/android-12/arch-arm/usr/include/math.h
@@ -523,10 +523,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-12/arch-mips/usr/include/math.h b/9/platforms/android-12/arch-mips/usr/include/math.h
index 5730cd4..16df2f1 100644
--- a/9/platforms/android-12/arch-mips/usr/include/math.h
+++ b/9/platforms/android-12/arch-mips/usr/include/math.h
@@ -523,10 +523,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-12/arch-x86/usr/include/math.h b/9/platforms/android-12/arch-x86/usr/include/math.h
index 5730cd4..16df2f1 100644
--- a/9/platforms/android-12/arch-x86/usr/include/math.h
+++ b/9/platforms/android-12/arch-x86/usr/include/math.h
@@ -523,10 +523,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-13/arch-arm/usr/include/math.h b/9/platforms/android-13/arch-arm/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-13/arch-arm/usr/include/math.h
+++ b/9/platforms/android-13/arch-arm/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-13/arch-mips/usr/include/math.h b/9/platforms/android-13/arch-mips/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-13/arch-mips/usr/include/math.h
+++ b/9/platforms/android-13/arch-mips/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-13/arch-x86/usr/include/math.h b/9/platforms/android-13/arch-x86/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-13/arch-x86/usr/include/math.h
+++ b/9/platforms/android-13/arch-x86/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-14/arch-arm/usr/include/math.h b/9/platforms/android-14/arch-arm/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-14/arch-arm/usr/include/math.h
+++ b/9/platforms/android-14/arch-arm/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-14/arch-mips/usr/include/math.h b/9/platforms/android-14/arch-mips/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-14/arch-mips/usr/include/math.h
+++ b/9/platforms/android-14/arch-mips/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-14/arch-x86/usr/include/math.h b/9/platforms/android-14/arch-x86/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-14/arch-x86/usr/include/math.h
+++ b/9/platforms/android-14/arch-x86/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-15/arch-arm/usr/include/math.h b/9/platforms/android-15/arch-arm/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-15/arch-arm/usr/include/math.h
+++ b/9/platforms/android-15/arch-arm/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-15/arch-mips/usr/include/math.h b/9/platforms/android-15/arch-mips/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-15/arch-mips/usr/include/math.h
+++ b/9/platforms/android-15/arch-mips/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-15/arch-x86/usr/include/math.h b/9/platforms/android-15/arch-x86/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-15/arch-x86/usr/include/math.h
+++ b/9/platforms/android-15/arch-x86/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-16/arch-arm/usr/include/math.h b/9/platforms/android-16/arch-arm/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-16/arch-arm/usr/include/math.h
+++ b/9/platforms/android-16/arch-arm/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-16/arch-mips/usr/include/math.h b/9/platforms/android-16/arch-mips/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-16/arch-mips/usr/include/math.h
+++ b/9/platforms/android-16/arch-mips/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-16/arch-x86/usr/include/math.h b/9/platforms/android-16/arch-x86/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-16/arch-x86/usr/include/math.h
+++ b/9/platforms/android-16/arch-x86/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-17/arch-arm/usr/include/math.h b/9/platforms/android-17/arch-arm/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-17/arch-arm/usr/include/math.h
+++ b/9/platforms/android-17/arch-arm/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-17/arch-mips/usr/include/math.h b/9/platforms/android-17/arch-mips/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-17/arch-mips/usr/include/math.h
+++ b/9/platforms/android-17/arch-mips/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-17/arch-x86/usr/include/math.h b/9/platforms/android-17/arch-x86/usr/include/math.h
index 7fe24b9..8f10936 100644
--- a/9/platforms/android-17/arch-x86/usr/include/math.h
+++ b/9/platforms/android-17/arch-x86/usr/include/math.h
@@ -528,10 +528,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-18/arch-arm/usr/include/math.h b/9/platforms/android-18/arch-arm/usr/include/math.h
index 042c98c..8a04986 100644
--- a/9/platforms/android-18/arch-arm/usr/include/math.h
+++ b/9/platforms/android-18/arch-arm/usr/include/math.h
@@ -533,10 +533,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-18/arch-mips/usr/include/math.h b/9/platforms/android-18/arch-mips/usr/include/math.h
index 042c98c..8a04986 100644
--- a/9/platforms/android-18/arch-mips/usr/include/math.h
+++ b/9/platforms/android-18/arch-mips/usr/include/math.h
@@ -533,10 +533,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-18/arch-x86/usr/include/math.h b/9/platforms/android-18/arch-x86/usr/include/math.h
index 042c98c..8a04986 100644
--- a/9/platforms/android-18/arch-x86/usr/include/math.h
+++ b/9/platforms/android-18/arch-x86/usr/include/math.h
@@ -533,10 +533,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-19/arch-arm/usr/include/math.h b/9/platforms/android-19/arch-arm/usr/include/math.h
index 042c98c..8a04986 100644
--- a/9/platforms/android-19/arch-arm/usr/include/math.h
+++ b/9/platforms/android-19/arch-arm/usr/include/math.h
@@ -533,10 +533,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-19/arch-mips/usr/include/math.h b/9/platforms/android-19/arch-mips/usr/include/math.h
index 042c98c..8a04986 100644
--- a/9/platforms/android-19/arch-mips/usr/include/math.h
+++ b/9/platforms/android-19/arch-mips/usr/include/math.h
@@ -533,10 +533,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-19/arch-x86/usr/include/math.h b/9/platforms/android-19/arch-x86/usr/include/math.h
index 042c98c..8a04986 100644
--- a/9/platforms/android-19/arch-x86/usr/include/math.h
+++ b/9/platforms/android-19/arch-x86/usr/include/math.h
@@ -533,10 +533,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-21/arch-arm/usr/include/math.h b/9/platforms/android-21/arch-arm/usr/include/math.h
index 0b711f7..9c5dd75 100644
--- a/9/platforms/android-21/arch-arm/usr/include/math.h
+++ b/9/platforms/android-21/arch-arm/usr/include/math.h
@@ -512,12 +512,12 @@ double	__builtin_expm1(double) __NDK_FPABI_MATH__;
 double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 /* See upstream bug http://llvm.org/bugs/show_bug.cgi?id=20958 */
 int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2;
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-21/arch-arm64/usr/include/math.h b/9/platforms/android-21/arch-arm64/usr/include/math.h
index 0b711f7..9c5dd75 100644
--- a/9/platforms/android-21/arch-arm64/usr/include/math.h
+++ b/9/platforms/android-21/arch-arm64/usr/include/math.h
@@ -512,12 +512,12 @@ double	__builtin_expm1(double) __NDK_FPABI_MATH__;
 double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 /* See upstream bug http://llvm.org/bugs/show_bug.cgi?id=20958 */
 int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2;
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-21/arch-mips/usr/include/math.h b/9/platforms/android-21/arch-mips/usr/include/math.h
index 0b711f7..9c5dd75 100644
--- a/9/platforms/android-21/arch-mips/usr/include/math.h
+++ b/9/platforms/android-21/arch-mips/usr/include/math.h
@@ -512,12 +512,12 @@ double	__builtin_expm1(double) __NDK_FPABI_MATH__;
 double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 /* See upstream bug http://llvm.org/bugs/show_bug.cgi?id=20958 */
 int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2;
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-21/arch-mips64/usr/include/math.h b/9/platforms/android-21/arch-mips64/usr/include/math.h
index 0b711f7..9c5dd75 100644
--- a/9/platforms/android-21/arch-mips64/usr/include/math.h
+++ b/9/platforms/android-21/arch-mips64/usr/include/math.h
@@ -512,12 +512,12 @@ double	__builtin_expm1(double) __NDK_FPABI_MATH__;
 double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 /* See upstream bug http://llvm.org/bugs/show_bug.cgi?id=20958 */
 int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2;
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-21/arch-x86/usr/include/math.h b/9/platforms/android-21/arch-x86/usr/include/math.h
index 0b711f7..9c5dd75 100644
--- a/9/platforms/android-21/arch-x86/usr/include/math.h
+++ b/9/platforms/android-21/arch-x86/usr/include/math.h
@@ -512,12 +512,12 @@ double	__builtin_expm1(double) __NDK_FPABI_MATH__;
 double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 /* See upstream bug http://llvm.org/bugs/show_bug.cgi?id=20958 */
 int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2;
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-21/arch-x86_64/usr/include/math.h b/9/platforms/android-21/arch-x86_64/usr/include/math.h
index 0b711f7..9c5dd75 100644
--- a/9/platforms/android-21/arch-x86_64/usr/include/math.h
+++ b/9/platforms/android-21/arch-x86_64/usr/include/math.h
@@ -512,12 +512,12 @@ double	__builtin_expm1(double) __NDK_FPABI_MATH__;
 double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 /* See upstream bug http://llvm.org/bugs/show_bug.cgi?id=20958 */
 int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2;
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-3/arch-arm/usr/include/math.h b/9/platforms/android-3/arch-arm/usr/include/math.h
index 0706a1f..aab0028 100644
--- a/9/platforms/android-3/arch-arm/usr/include/math.h
+++ b/9/platforms/android-3/arch-arm/usr/include/math.h
@@ -523,10 +523,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-4/arch-arm/usr/include/math.h b/9/platforms/android-4/arch-arm/usr/include/math.h
index 0706a1f..aab0028 100644
--- a/9/platforms/android-4/arch-arm/usr/include/math.h
+++ b/9/platforms/android-4/arch-arm/usr/include/math.h
@@ -523,10 +523,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-5/arch-arm/usr/include/math.h b/9/platforms/android-5/arch-arm/usr/include/math.h
index 0706a1f..aab0028 100644
--- a/9/platforms/android-5/arch-arm/usr/include/math.h
+++ b/9/platforms/android-5/arch-arm/usr/include/math.h
@@ -523,10 +523,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-8/arch-arm/usr/include/math.h b/9/platforms/android-8/arch-arm/usr/include/math.h
index 0706a1f..aab0028 100644
--- a/9/platforms/android-8/arch-arm/usr/include/math.h
+++ b/9/platforms/android-8/arch-arm/usr/include/math.h
@@ -523,10 +523,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-9/arch-arm/usr/include/math.h b/9/platforms/android-9/arch-arm/usr/include/math.h
index 5730cd4..16df2f1 100644
--- a/9/platforms/android-9/arch-arm/usr/include/math.h
+++ b/9/platforms/android-9/arch-arm/usr/include/math.h
@@ -523,10 +523,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-9/arch-mips/usr/include/math.h b/9/platforms/android-9/arch-mips/usr/include/math.h
index 5730cd4..16df2f1 100644
--- a/9/platforms/android-9/arch-mips/usr/include/math.h
+++ b/9/platforms/android-9/arch-mips/usr/include/math.h
@@ -523,10 +523,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
diff --git a/9/platforms/android-9/arch-x86/usr/include/math.h b/9/platforms/android-9/arch-x86/usr/include/math.h
index 5730cd4..16df2f1 100644
--- a/9/platforms/android-9/arch-x86/usr/include/math.h
+++ b/9/platforms/android-9/arch-x86/usr/include/math.h
@@ -523,10 +523,10 @@ double	__builtin_fma(double, double, double) __NDK_FPABI_MATH__;
 double	__builtin_hypot(double, double) __NDK_FPABI_MATH__;
 int	__builtin_ilogb(double) __NDK_FPABI_MATH__ __pure2;
 /* int	__builtin_isinf(double) __NDK_FPABI_MATH__ __pure2; */
-#if !defined(__clang__) || __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+#if !defined(__clang__)
 int	__builtin_isnan(double) __NDK_FPABI_MATH__ __pure2;
 #else
-/* clang < 3.7 has faulty prototype for __builtin_isnan */
+/* clang has faulty prototype for __builtin_isnan */
 #endif
 double	__builtin_lgamma(double) __NDK_FPABI_MATH__;
 long long __builtin_llrint(double) __NDK_FPABI_MATH__;
-- 
1.9.1

